yoyodyne/models/modules/rnn.py

```diff
@@ -49,46 +51,71 @@ def forward(self, source: data.PaddedTensor) -> base.ModuleOutput:
                 for source, of shape B x seq_len x 1.
 
         Returns:
-            base.ModuleOutput: encoded timesteps, and LSTM h0 and c0 cells.
+            base.ModuleOutput.
         """
         embedded = self.embed(source.padded)
         # Packs embedded source symbols into a PackedSequence.
         packed = nn.utils.rnn.pack_padded_sequence(
-            embedded, source.lengths(), batch_first=True, enforce_sorted=False
+            embedded,
+            source.lengths(),
+            batch_first=True,
+            enforce_sorted=False,
         )
-        # -> B x seq_len x encoder_dim, (h0, c0).
-        packed_outs, (H, C) = self.module(packed)
+        # -> B x seq_len x encoder_dim, hiddens
+        packed_outs, hiddens = self.module(packed)
         encoded, _ = nn.utils.rnn.pad_packed_sequence(
             packed_outs,
             batch_first=True,
             padding_value=special.PAD_IDX,
             total_length=None,
         )
-        return base.ModuleOutput(encoded, hiddens=(H, C))
+        return base.ModuleOutput(encoded, hiddens)
 
-    def get_module(self) -> nn.LSTM:
-        return nn.LSTM(
+    @property
+    def output_size(self) -> int:
+        return self.hidden_size * self.num_directions
+
+
+class GRUEncoder(RNNEncoder):
```

Comment:
Very minor, but if we wanted to simplify, the GRU and LSTM Encoder can just be an RNN encoder with a data factory call to nn.X. Bit on the fence if that's good for abstraction or just trying to be annoyingly compact.